Adding ConfigClass for HTTPX

mattjacksoncello opened the pull request:
This PR is to implement a config class to allow more control over httpx for things like certificate verification.

I have also merged in the changes from #92 

Note: I have used Claude to help with the changes. I have reviewed everything but I am not the strongest coder so there may be mistakes that I have missed

AndyTitu commented on src/onepasswordconnectsdk/__init__.py:
```diff
@@ -4,12 +4,5 @@
 from onepasswordconnectsdk.config import load_dict
 from onepasswordconnectsdk.client import new_client
 from onepasswordconnectsdk.client import new_client_from_environment
-
-__all__ = [
-    "client",
-    "load",
-    "load_dict",
-    "models",
-    "new_client",
-    "new_client_from_environment"
-]
```
Won't deleting this block avoid these components from being properly exported from this package?

AndyTitu commented on src/tests/test_client_vaults.py:
```diff
@@ -1,13 +1,18 @@
 TOKEN = "jwt_token"
-SS_CLIENT = client.new_client(HOST, TOKEN)
-SS_CLIENT_ASYNC = client.new_client(HOST, TOKEN, True)
+# Create clients using new config pattern
+client_config = ClientConfig(url=HOST, token=TOKEN)
+SS_CLIENT = client.Client(client_config)
+
+async_config = AsyncClientConfig(url=HOST, token=TOKEN)
+SS_CLIENT_ASYNC = async_client.AsyncClient(async_config)
```
@mattjacksoncello could you

AndyTitu commented on src/onepasswordconnectsdk/client.py:
```diff
@@ -381,34 +388,34 @@ def sanitize_for_serialization(self, obj):
         return self.serializer.sanitize_for_serialization(obj)
 
 
-def new_client(url: str, token: str, is_async: bool = False) -> Union[AsyncClient, Client]:
+def new_client(url: str, token: str, **kwargs) -> Client:
```
This change is not backwards compatible unfortunately, and other existing user scripts will break if we ever released this and those users updated to this new version. To support this new separation we'd have to release a new major version of the SDK, namely 2.0, and I'm not sure it's worth it only to separate the client initialisation function. 

Don't get me wrong, I do agree that this would probably be a better pattern, it's just that I don't believe that this improvement alone is worth releasing a 2.0. Can we maybe revert this client separation and only keep the client config part of this PR?

mattjacksoncello commented:
Thanks for the review Andy, will refactor the code

mattjacksoncello commented:
@AndyTitu,

I've updated the PR to focus solely on the config class changes as requested. Here are the key updates:

1. Documentation:
   - Added a new section in USAGE.md documenting the primary use case for CA certificate verification with Connect server
   
2. Testing:
   - Created a new test file specifically for the config class implementation
   - Looking for feedback on test coverage and potential edge cases
   
3. Examples:
   - Added a new example demonstrating CA file usage in practice

Please review these changes and let me know if any adjustments are needed.

mattjacksoncello commented:
Hi @AndyTitu 

Any updates on this?

Thanks

mattjacksoncello commented:
Hi @AndyTitu 
When do you think you will have time to review this? I would like to use the production version if possible.
Thanks

AndyTitu commented:
@mattjacksoncello sorry for missing notifications on this. I'm on it now.

AndyTitu commented on src/onepasswordconnectsdk/config.py:
```diff
@@ -16,7 +19,48 @@
+            cafile (Optional[str]): Path to CA certificate file for SSL verification
+            **kwargs: Additional httpx client options
+        """
+        self.cafile = cafile
```
```suggestion
        self.ca_file = ca_file
```

AndyTitu reviewed (approved):
Code looks good to me apart from some small suggestions, thank you for the contribution! ❤️

AndyTitu commented:
@mattjacksoncello some tests are failing for the last changes.

mattjacksoncello commented:
Hi Andy,
I have renamed all instances of cafile to ca_file. Test are now working
Cheers!

mattjacksoncello commented:
@AndyTitu Could you have another look at this please

mattjacksoncello commented:
Hi @AndyTitu  
when will this be integrated into a release?
